### PR TITLE
Handle inconsistent local roles

### LINKED ISSUE #3592: Behandeln von inkonsistenten lokalen Rollen beim schützen eines Dossiers
Im PR: https://github.com/4teamwork/opengever.core/pull/3580 wurde die Funktion zum Schützen eines Dossiers implementiert.

Momentan werden nach dem speichern eines Dossiers die lokalen Rollen mit den unter "Dossier schützen" angegebenen Rollen ersetzt.

Es ist jedoch möglich, dass die lokalen Rollen z.B. durch das Hinzufügen von Aufgaben oder Anträgen vom System gesetzt wurden.

Damit der Benutzer beim nächsten Bearbeiten des Dokuments weiss, dass er die Rollen nach dem Speichern überschreiben wird, soll eine Warnmeldung angezeigt werden.

- [x] Warnmeldung im Edit-Form anzeigen, wenn die lokalen Rollen unter "Schützen" nich den effektiven lokalen Rollen entsprechen.

### PR DESCRIPTION
**Pull request by elioschmutz**
This PR handles inconsistent local roles:

screen1

closes #3592

### CODE REVIEW

File: opengever/base/browser/resources/keywordwidget.js [hunk 1]
```diff
@@ -53,8 +53,43 @@
             }
         };
     }
+    function showConsistencyWarning(text) {
+        if ($(".protectDossierConsistencyMessage").length > 0) {
+            // Consistency waring already exists
```
Comment (phgross) [inline opengever/base/browser/resources/keywordwidget.js]: warning

File: opengever/dossier/browser/forms.py [hunk 2]
```diff
@@ -101,9 +98,10 @@ def label(self):
     def applyChanges(self, data):
         changes = super(DossierEditForm, self).applyChanges(data)
 
-        # CUSTOM: Handle dossier protection after successfully adding
-        # the dossier
-        if IProtectDossierMarker.providedBy(self.context):
+        # CUSTOM: Handle dossier protection after editiing the dossier.
```
Comment (phgross) [inline opengever/dossier/browser/forms.py]: editing

File: sources.cfg [hunk 3]
```diff
@@ -10,6 +10,7 @@ development-packages =
   collective.js.timeago
 # https://github.com/4teamwork/plonetheme.teamraum/pull/595
   plonetheme.teamraum
+  ftw.keywordwidget
```
Comment (phgross) [inline sources.cfg]: please make another release ...

File: opengever/dossier/browser/protect_dossier.py [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+from Products.Five import BrowserView
+from zope.i18n import translate
+import json
+
+
+class CheckProtectDossierConsistency(BrowserView):
+    """
+    """
+    def __call__(self):
+        response = self.request.response
```
Comment (phgross) [inline opengever/dossier/browser/protect_dossier.py]: Could you use opengever.base.response.JSONResponse here ?

Comment (elioschmutz) [reply opengever/dossier/browser/protect_dossier.py]: 👍 yes, I can. But the client side implementation of `MessageFactory` does not what we want. 

I'll still render the message as a default portal message. The MessageFactory just notifies the user about something. The notification will disappear after a while. We could adjust the implementation to allow own settings to get a static message. But I think, the default portal message is well enough. What do you think?

Comment (phgross) [reply opengever/dossier/browser/protect_dossier.py]: Hmm ok. No then it makes no sense ...

**Comment by elioschmutz** [comment]
@phgross 👍 updated the PR. Take a look at my new commits.

**Pull request merged**